sky/provision/kubernetes/instance.py

```diff
@@ -253,6 +253,31 @@ def _run_command_on_pods(node_name, node_namespace, command):
         _request_timeout=kubernetes.API_TIMEOUT)
     return cmd_output
 
+def _run_command_on_pods_with_logs(node_name, node_namespace, command, command_list):
```

Comment:
I'm thinking can we merge this method with `_run_command_on_pods`?

The new signature can be:
`def _run_command_on_pods(node_name: str, node_namespace: str, command_list: List[str],  shell_command: List[str] = ['/bin/sh'], stream_logs: bool = False, operation_name: Optional[str] = None):`

if stream_logs is set to True, we use your streaming codepath below and use logger.info/logger.warning to log commands and outputs.

Reply:
Also since this method will be getting complicated, it will be nice to add a docstr : )